### ISSUE TITLE
Translate complete websites?

### ISSUE THREAD

**Issue by wuniversales** [uav4geo/LibreTranslate#14]
How to translate full websites?

It would be good to put examples to translate webs into any other language.

**Comment by pierotofy** [comment]
One would probably need to develop some sort of browser extension. Would be a cool idea for a project!

**Comment by Extarys** [comment]
Would it be possible to do like Google and when someone enters a link, the translation side shows the same link, when clicked on it the webpage is translated?

Would it be better to translate the page server-side or client-side?

Also see https://github.com/uav4geo/LibreTranslate/issues/55

**Comment by the-red-lily** [comment]
It seems that libretranslate now has support for HTML. How possible would it be to now support adding website translation to the web interface?

**Comment by dingedi** [comment]
yes libretranslate supports html translation thanks to [https://github.com/argosopentech/translate-html](https://github.com/argosopentech/translate-html)   
I think it's a little bit complex if we want to achieve a result like the translation of websites from google translate but it is doable and it would be a nice addition ( [example](https://github-com.translate.goog/LibreTranslate/LibreTranslate/issues/14?_x_tr_sl=en&_x_tr_tl=es&_x_tr_hl=fr&_x_tr_pto=wapp) )
do you want to open a PR?

**Comment by the-red-lily** [comment]
I can take a look

**Comment by satoshinotdead** [comment]
That should help us to stop using DeepL, for sure :)

**Comment by satoshinotdead** [comment]
There is some progress here?

Translating entire websites should motivate more users to integrate Libretranslate to their projects!

**Comment by sommmen** [comment]
I'm interested in translating emails for which i believe this would also be appropriate.

**Comment by seadropx** [comment]
This feature will be great if fulled. Or I wonder how to use API in the[ immersive-translate ](https://immersivetranslate.com/) extension in browser, for the immersive-translate extension can translate website by many translation services such as google, Microsoft et al.

**Comment by dnkcz** [comment]
why would you use that instead of using google translation of web pages that is already there?
